docker-entrypoint/docker-entrypoint.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+	//For testing purposes
+	//"k8s.io/kubernetes/pkg/client/restclient"
+
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	WAITING  = "waiting"
+	READY    = "ready"
+	RUNNING  = "Running"
+	INTERVAL = 2
+)
+
+var (
+	//"Info logger""
+	Info *log.Logger
+	//"Error logger"
+	Error *log.Logger
+)
+
+type dependency interface {
```

Comment:
Call the interface more generally, like `Resolvable` and expose `GetType()` and `IsResolved()`. Because it is more general, parameters should then come from the object.